public/components/custom_panels/forms/add_saved_visualizations.tsx

```diff
@@ -0,0 +1,124 @@
+import { EuiSpacer, EuiFormRow, EuiSelect, EuiSuperDatePicker, EuiButtonEmpty, EuiButton, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React, { useState } from 'react';
+import { VisualizationType } from '../../../../common/constants/custom_panels';
+import PPLService from '../../../services/requests/ppl';
+
+//NOTE: working this module is TBD after work on storing ppl visualizations in index is complete
+
+/*
+ * "AddSavedVisualizations" component to add saved visualizations
+ *
+ * closeVizWindow: function to close "add visualization" window
+ * pplService: PPLService Requestor 
+ * panelVisualizations: panelVisualizations object 
+ * setPanelVisualizations: Setter for panelVisualizations object
+ * setToast: Create Toast function 
+ */
+
+type Props = {
+  closeVizWindow: () => void;
+  pplService: PPLService;
+  panelVisualizations: VisualizationType[];
+  setPanelVisualizations: React.Dispatch<React.SetStateAction<VisualizationType[]>>;
+  setToast: (title: string, color?: string, text?: string) => void;
+};
+
+export const AddSavedVisualizations = ({
+  closeVizWindow,
+  pplService,
+  panelVisualizations,
+  setPanelVisualizations,
+  setToast,
+}: Props) => {
+  const [newVisualizationTitle, setNewVisualizationTitle] = useState('');
+  const [newVisualizationType, setNewVisualizationType] = useState('bar');
+  const [pplQuery, setPPLQuery] = useState('');
+  const [previewData, setPreviewData] = useState([]);
+  const [previewArea, setPreviewArea] = useState(<></>);
+  const [showPreviewArea, setShowPreviewArea] = useState(false);
+  const [previewIconType, setPreviewIconType] = useState('arrowRight');
+  const [previewLoading, setPreviewLoading] = useState(false);
+  const [isPreviewError, setIsPreviewError] = useState('');
+
+  // DateTimePicker States
+  const [recentlyUsedRanges, setRecentlyUsedRanges] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [start, setStart] = useState('now-30m');
+  const [end, setEnd] = useState('now');
+
+  const onTimeChange = ({ start, end }) => {
+    const recentlyUsedRange = recentlyUsedRanges.filter((recentlyUsedRange) => {
+      const isDuplicate = recentlyUsedRange.start === start && recentlyUsedRange.end === end;
+      return !isDuplicate;
+    });
+    recentlyUsedRange.unshift({ start, end });
+    setStart(start);
+    setEnd(end);
+    setRecentlyUsedRanges(
+      recentlyUsedRange.length > 10 ? recentlyUsedRange.slice(0, 9) : recentlyUsedRange
+    );
+    setIsLoading(true);
+    startLoading();
+    console.log('time change', start, end);
```

Comment:
could you remove the console.log? also this `onTimeChange` looks identical in other places, is it possible to move it to utils.tsx to reduce duplicates and for easier unit tests?

Reply:
Indeed this is duplicating code. Will move this to utils.